### DISCUSSION TITLE
af packet ring size

### DISCUSSION THREAD

**Discussion by Stephen-Ridgway** [Security-Onion-Solutions/securityonion#4600] [Unsupported Versions] [answered]
Hi!

How can I set the af packet ring size for a specific sensor? I'm seeing Suricata packet loss (about 5%) and I'd like to try increasing the ring size to see if this helps.

Thanks!
Stephen

**Comment by Stephen-Ridgway** [comment, marked as answer]
Found it!

suriringsize: 

in the sensor's pillar file

**Comment by corycandia** [comment]
@Stephen-Ridgway,

Couple of questions:

1. How did you find this setting?

I understand the basic concept of SALT passing the structured settings that get parsed on the other end, but that's about it.  You might have to explain it like I'm 5.  

Since you had the answer, I searched it on github (suriringsize) and saw the file (securityonion/salt/suricata/afpacket.map.jinja) has it in there, but I would not have been able to figure that out otherwise.  How did you?

2. It looks like the default is 5000?  Docs say 5000 packets per instance can sit waiting to be processed before drop if I understand correctly.  Mine is set for three instances right now (six core VM):

`zeek_lbprocs: 3`
`suriprocs: 3`

So that should mean 15,000 packets can sit waiting to be processed before drop, right?  Looks like i'm still dropping.
![image](https://user-images.githubusercontent.com/88662234/129126560-bae01270-1899-46e6-8b8a-701eaabf5c77.png)

I am looking to eliminate that couple of percent.  What did you change your environment to?  Just keep upping it until no loss?

3. Other than playing with process instance count and suriringsize, do we adjust any other buffers anymore?  I remember the 16.04 had ring size and buffer you could run up to 256MB / 65K respectively.

I'd rather not BPF out the SSL traffic, but I guess the connections are still tracked.  I pondered how stupid it would be to try and import my SSL proxy signing cert and try implementing un-encrypting the packets, but if SO doesn't natively support it, then not.

**Reply by TOoSmOotH** [reply]
Looking at your graph Zeek and Suricata are fine. Your PCAP packet loss is most likely due to IO constraints. Zeek Capture loss could be due to async routing or your span dropping packets. There isn't anything you can "tune" on the machine to reduce capture loss.
